File: app/src/main/java/com/dkarakaya/songify/SongAdapter.java
package com.dkarakaya.songify;

import android.content.Context;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

public class SongAdapter extends RecyclerView.Adapter<SongAdapter.SongHolder> {

    private ArrayList<SongInfo> _songs;
    private Context context;
    private OnItemClickListener mOnItemClickListener;

    public SongAdapter(Context context, ArrayList<SongInfo> songs) {
        this.context = context;
        this._songs = songs;
    }

    public interface OnItemClickListener {
        void onItemClick(View view, SongInfo obj, int position);
    }

    public void setOnItemClickListener(final OnItemClickListener mItemClickListener) {
        this.mOnItemClickListener = mItemClickListener;
    }


    @Override
    public SongHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View myView = LayoutInflater.from(context).inflate(R.layout.row_song,viewGroup,false);
        return new SongHolder(myView);
    }

    @Override
    public void onBindViewHolder(final SongHolder songHolder, final int i) {
        final SongInfo s = _songs.get(i);
        songHolder.tvSongName.setText(_songs.get(i).getSongTitle());
        songHolder.tvSongArtist.setText(_songs.get(i).getArtistName());
        songHolder.tvSongDuration.setText(_songs.get(i).getSongDuration());
        songHolder.cvPlay.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mOnItemClickListener != null) {
                    mOnItemClickListener.onItemClick(v, s, i);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return _songs.size();
    }

    public class SongHolder extends RecyclerView.ViewHolder {
        TextView tvSongName,tvSongArtist, tvSongDuration;
        CardView cvPlay;
        public SongHolder(View itemView) {
            super(itemView);
            tvSongName = itemView.findViewById(R.id.tvSongName);
            tvSongArtist = itemView.findViewById(R.id.tvArtistName);
            tvSongDuration = itemView.findViewById(R.id.tvSongDuration);
            cvPlay=itemView.findViewById(R.id.cvPlay);
        }
    }
}

File: app/src/main/java/com/dkarakaya/songify/SongAdapter.kt
package com.dkarakaya.songify

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.dkarakaya.songify.SongAdapter.SongHolder

class SongAdapter(private val context: Context, private val _songs: MutableList<SongInfo>) : RecyclerView.Adapter<SongHolder>() {
    private var mOnItemClickListener: OnItemClickListener? = null

    interface OnItemClickListener {
        fun onItemClick(view: View?, obj: SongInfo?, position: Int)
    }

    fun setOnItemClickListener(mItemClickListener: OnItemClickListener) {
        mOnItemClickListener = mItemClickListener
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): SongHolder {
        val myView = LayoutInflater.from(context).inflate(R.layout.row_song, viewGroup, false)
        return SongHolder(myView)
    }

    override fun onBindViewHolder(songHolder: SongHolder, i: Int) {
        val s = _songs[i]
        songHolder.tvSongName.text = _songs[i].songTitle
        songHolder.tvSongArtist.text = _songs[i].artistName
        songHolder.tvSongDuration.text = _songs[i].songDuration
        songHolder.cvPlay.setOnClickListener { v ->
            if (mOnItemClickListener != null) {
                mOnItemClickListener!!.onItemClick(v, s, i)
            }
        }
    }

    override fun getItemCount(): Int {
        return _songs.size
    }

    inner class SongHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvSongName: TextView
        var tvSongArtist: TextView
        var tvSongDuration: TextView
        var cvPlay: CardView

        init {
            tvSongName = itemView.findViewById(R.id.tvSongName)
            tvSongArtist = itemView.findViewById(R.id.tvArtistName)
            tvSongDuration = itemView.findViewById(R.id.tvSongDuration)
            cvPlay = itemView.findViewById(R.id.cvPlay)
        }
    }

}
